### PR TITLE
Document the --extended-logging variable

### PR DESCRIPTION
**Pull request by mburke5678**
https://bugzilla.redhat.com/show_bug.cgi?id=1728006

### CODE REVIEW

File: install_config/router/default_haproxy_router.adoc [hunk 1]
```diff
@@ -98,19 +98,21 @@ not exist, run the following to create a router:
...
-$ oc adm router <router_name> --replicas=<number> --service-account=router
+$ oc adm router <router_name> --replicas=<number> --service-account=router --extended-logging=true
 ----
 endif::[]
 
 `--replicas` is usually `1` unless a
 xref:../../admin_guide/high_availability.adoc#[high availability] configuration
 is  being created.
 
+`--extended-logging=true` configures the router to forward access logs to your EFK stack.
```
Comment (anpingli) [inline install_config/router/default_haproxy_router.adoc:114]: ---> `--extended-logging=true`  enable a cartridge container to forward access logs to its standard output.

Comment (anpingli) [reply install_config/router/default_haproxy_router.adoc:114]: @lihongan  Could you confirm that is a support feature or a workaround?

Comment (lihongan) [reply install_config/router/default_haproxy_router.adoc:114]: I don't think so, I just create another sidecar container that running rsyslogd.

File: install_config/router/default_haproxy_router.adoc [hunk 2]
```diff
@@ -166,6 +168,50 @@ $ oc adm router --dry-run -o yaml --service-account=router
...
+
+For example, the following is a router configuration created using `--extended-logging=true`: 
+----
+# oc get pod test-router-1-xhdb9 -o yaml
+apiVersion: v1
+kind: Pod
+spec:
+  containers:
+  - env:
+    ....
```
Comment (lihongan) [inline install_config/router/default_haproxy_router.adoc:191]: And the below env is added to router container with option `--extended-logging=true`, so better to add it here.
```
    - name: ROUTER_SYSLOG_ADDRESS
      value: /var/lib/rsyslog/rsyslog.sock
```

File: install_config/router/default_haproxy_router.adoc [hunk 3]
```diff
@@ -166,6 +168,50 @@ $ oc adm router --dry-run -o yaml --service-account=router
...
+   - /etc/rsyslog/rsyslog.conf
+   image: registry.redhat.io/openshift3/ose-haproxy-router:v3.11.188
+   imagePullPolicy: IfNotPresent
+   name: syslog
+----
+<1> The parameter added a new container to the router. In the container, 
+the rsyslog process is running as: 
+`/sbin/rsyslogd -n -i /tmp/rsyslog.pid -f /etc/rsyslog/rsyslog.conf`.
+
+Use the following command to view the access logs:
```
Comment (lihongan) [inline install_config/router/default_haproxy_router.adoc:207]: See also https://docs.openshift.com/container-platform/3.11/admin_guide/router.html#viewing-logs
The default log level is warning and should be set to `info` or `debug` if you want to view the access logs.
`oc set env dc/test-router ROUTER_LOG_LEVEL=info`.

File: install_config/router/default_haproxy_router.adoc [hunk 4]
```diff
@@ -166,6 +168,50 @@ $ oc adm router --dry-run -o yaml --service-account=router
...
+
+----
+# oc logs -f <pod-name> -c syslog
```
Comment (lihongan) [inline install_config/router/default_haproxy_router.adoc:210]: One example of access log:
```
2020-04-14T03:05:36.629527+00:00 test-311-node-1 haproxy[43]: 10.0.151.166:59594 [14/Apr/2020:03:05:36.627] fe_no_sni~ be_secure:openshift-console:console/pod:console-b475748cb-t6qkq:console:10.128.0.5:8443 0/0/1/1/2 200 393 - - --NI 2/1/0/1/0 0/0 "HEAD / HTTP/1.1"
2020-04-14T03:05:36.633024+00:00 test-311-node-1 haproxy[43]: 10.0.151.166:59594 [14/Apr/2020:03:05:36.528] public_ssl be_no_sni/fe_no_sni 95/1/104 2793 -- 1/1/0/0/0 0/0

```

File: install_config/router/default_haproxy_router.adoc (hunk 1 above)
Comment (lihongan) [reply install_config/router/default_haproxy_router.adoc:114]: It is not related to `EFK stack`,  should be `configures the router to forward HAProxy log to the syslog container.`

File: install_config/router/default_haproxy_router.adoc [hunk 5]
```diff
@@ -166,6 +168,50 @@ $ oc adm router --dry-run -o yaml --service-account=router
 ----
 endif::[]
 
+[[deploy-router-access-logs]]
+Configuring the Router to Forward Access Logs::
+
+// tag::accesslogs[]
+You can configure {product-title} to send socket messages, or access logs, 
+to your EFK stack. The `--extended-logging` parameter appends an rsyslog
```
Comment (lihongan) [inline install_config/router/default_haproxy_router.adoc:176]: Same to above, `to the syslog container` not `EFK stack`.

File: install_config/router/default_haproxy_router.adoc (hunk 2 above)
Comment (mburke5678) [reply install_config/router/default_haproxy_router.adoc:193]: Outdated

**Comment by mburke5678** [comment]
@lihongan Thank you for your comments. I made the changes as you suggested.

**Comment by lihongan** [comment]
LGTM. Thank you @mburke5678

**Comment by mburke5678** [comment]
@openshift/team-documentation PTAL

File: install_config/router/default_haproxy_router.adoc [hunk 6]
```diff
@@ -98,19 +98,22 @@ not exist, run the following to create a router:
...
-$ oc adm router <router_name> --replicas=<number> --service-account=router
+$ oc adm router <router_name> --replicas=<number> --service-account=router --extended-logging=true
 ----
 endif::[]
 
 `--replicas` is usually `1` unless a
 xref:../../admin_guide/high_availability.adoc#[high availability] configuration
 is  being created.
 
+`--extended-logging=true` configures the router to forward logs generated by HAProxy
```
Comment (kalexand-rh) [inline install_config/router/default_haproxy_router.adoc:114]: s/generated/that are generated

File: install_config/router/default_haproxy_router.adoc [hunk 7]
```diff
@@ -166,6 +169,66 @@ $ oc adm router --dry-run -o yaml --service-account=router
 ----
 endif::[]
 
+[[deploy-router-access-logs]]
+Configuring the Router to Forward HAProxy Logs::
+
+You can configure the router to forward logs generated by HAProxy
```
Comment (kalexand-rh) [inline install_config/router/default_haproxy_router.adoc:175]: s/generated/that are generated

File: install_config/router/default_haproxy_router.adoc [hunk 8]
```diff
@@ -166,6 +169,66 @@ $ oc adm router --dry-run -o yaml --service-account=router
...
+
+You can configure the router to forward logs generated by HAProxy 
+to an rsyslog sidecar container. The `--extended-logging=true` parameter 
+appends the syslog container to forward HAProxy logs to standard output. 
+
+----
+$ oc adm router --extended-logging=true
+----
+
+For example, the following is the configuration for a router
```
Comment (kalexand-rh) [inline install_config/router/default_haproxy_router.adoc:183]: I'd s/For example, the following/The following example

> :no_entry_sign: following n
Do not use as a noun except to refer to a group of followers, adherents, or partisans. For example, do not
write “Do the following” or “The functions include the following” as lead-in wording to a list. Use
“following” only as an adjective, as in “Complete the following steps” or “The adapter has the following
functions.”

File: install_config/router/default_haproxy_router.adoc [hunk 9]
```diff
@@ -166,6 +169,66 @@ $ oc adm router --dry-run -o yaml --service-account=router
...
+You can configure the router to forward logs generated by HAProxy 
+to an rsyslog sidecar container. The `--extended-logging=true` parameter 
+appends the syslog container to forward HAProxy logs to standard output. 
+
+----
+$ oc adm router --extended-logging=true
+----
+
+For example, the following is the configuration for a router 
+created using `--extended-logging=true`:
```
Comment (kalexand-rh) [inline install_config/router/default_haproxy_router.adoc:184]: Maybe s/created using/that uses

File: install_config/router/default_haproxy_router.adoc [hunk 10]
```diff
@@ -166,6 +169,66 @@ $ oc adm router --dry-run -o yaml --service-account=router
...
+   - -i
+   - /tmp/rsyslog.pid
+   - -f
+   - /etc/rsyslog/rsyslog.conf
+   image: registry.redhat.io/openshift3/ose-haproxy-router:v3.11.188
+   imagePullPolicy: IfNotPresent
+   name: syslog
+----
+<1> The parameter creates a socket file for the logs.
+<2> The parameter adds a new container to the router. In the container,
```
Comment (kalexand-rh) [inline install_config/router/default_haproxy_router.adoc:212]: ~new~

File: install_config/router/default_haproxy_router.adoc [hunk 11]
```diff
@@ -166,6 +169,66 @@ $ oc adm router --dry-run -o yaml --service-account=router
...
+   - -n
+   - -i
+   - /tmp/rsyslog.pid
+   - -f
+   - /etc/rsyslog/rsyslog.conf
+   image: registry.redhat.io/openshift3/ose-haproxy-router:v3.11.188
+   imagePullPolicy: IfNotPresent
+   name: syslog
+----
+<1> The parameter creates a socket file for the logs.
```
Comment (kalexand-rh) [inline install_config/router/default_haproxy_router.adoc:211]: Are these parameters or sections or stanzas?

File: install_config/router/default_haproxy_router.adoc [hunk 12]
```diff
@@ -166,6 +169,66 @@ $ oc adm router --dry-run -o yaml --service-account=router
...
+Use the following commands to view the HAProxy logs:
+
+----
+$ oc set env dc/test-router ROUTER_LOG_LEVEL=info <1> 
+$ oc logs -f <pod-name> -c syslog <2>
+----
+<1> Set the log level to `info` or `debug`. The default is `warning`.
+<2> Specify the name of the router pod to view the logs.
+
+The HAProxy logs appear similar to the following:
```
Comment (kalexand-rh) [inline install_config/router/default_haproxy_router.adoc:225]: Maybe "The HAProxy logs take the following form:"

**Review by kalexand-rh — COMMENTED** [review]
I have a few picks, but it largely LGTM.

File: install_config/router/default_haproxy_router.adoc (hunk 10 above)
Comment (mburke5678) [reply install_config/router/default_haproxy_router.adoc]: Changed to _The `--extended-logging=true` parameter..._ to highlight that the --extended-logging=true` added the socket file and container.

**Pull request merged**